utils/fixed_decimal/src/decimal.rs

```diff
@@ -1180,19 +1201,25 @@ impl FromStr for FixedDecimal {
     fn from_str(input_str: &str) -> Result<Self, Self::Err> {
         // input_str: the input string
         // no_sign_str: the input string when the sign is removed from it
-        // Check if the input string is "" or "-"
-        if input_str.is_empty() || input_str == "-" {
+        if input_str.is_empty() {
             return Err(Error::Syntax);
         }
         let input_str = input_str.as_bytes();
         #[allow(clippy::indexing_slicing)] // TODO(#1668) Clippy exceptions need docs or fixing.
```

Comment:
Nit, here and 5 lines below: Fix the `TODO(#1668)` while you're in here. You can say:

```
// The string is not empty
```